torch/functional.py

```diff
@@ -200,12 +199,11 @@ def lu_unpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
         >>> torch.norm(A_ - A)
         tensor(2.9802e-08)
     """
-    if not torch.jit.is_scripting():
```

Comment:
I wasn't sure the tests would fail but they did...

The torch JIT isn't capable of `__torch_function__`, so it bails on things in here. Writing the `if` statement like this makes it so the JIT can handle things properly.

Reply:
@suo how would you feel if we add a special case to TorchScript to identify blocks like this and treat them specially in the same way as `is_scripting()`? We can change the precise wording of the condition here as necessary, but with the constraint that eager mode perf has to be as fast as possible.

Reply:
I think that's reasonable, given that this block is hit from everything ever. We can define a `_has_torch_function_noscript` or something equally unlikely for users to use and have that is statically resolved to false in TS. cc @gmagogsfm @SplitInfinity, what do you think?

Reply:
Is the goal here to merge the two predicates of these if statements into one?

Reply:
The goal here is to reduce eager overhead. Merging all of these predicates together into a single predicate would certainly work, although we still have to be careful not to accidentally start allocating tuples to make this call.

Reply:
Tests are failing because the schema of `has_torch_function` is not valid. (It should be `Any[]`, but that's not a thing.) Some initial grep-ing indicates that the solution is probably to make a kind of `SugaredValue` which swallows arguments and then register `has_torch_function` as taking no args, but I figured I should check with you folks to see if there is a better way.

Reply:
Actually, it looks like `Any` is allowed, so I'm good.